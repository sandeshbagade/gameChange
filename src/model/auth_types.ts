import { Persistor } from 'redux-persist/es/types';
export interface Auth {
  isAuthenticated: boolean;
  isLoading: boolean;
  user: Record<string, any>;
  dicomPreview: {
    image: { file: File; preview: string; hide: boolean };
    tags: {};
    download_id: string;
  };
  pdfPreview: {
    pdf: string;
    hide: boolean;
    download_id: string;
  };
  batchPreview: {
    batch: {};
    hide: boolean;
  };
  batchProcess: { zipfile: [] };
  reviewTags: {
    currentRecord: number;
    totalRecord: number;
    review_tags: { plane: string; pathology: string; record_status: string; request_report: string };
  };
  token: any;
  alert: { message: string; status: boolean };
}
// eslint-disable-next-line @typescript-eslint/class-name-casing
export interface storeState {
  auth: any;
  persistor: Persistor;
}
export enum AuthActions {
  USER_LOADING = 'USER_LOADING',
  USER_LOADED = 'USER_LOADED',
  AUTH_ERROR = 'AUTH_ERROR',
  LOGIN_SUCCESS = 'LOGIN_SUCCESS',
  LOGIN_FAIL = 'LOGIN_FAIL',
  LOGOUT_SUCCESS = 'LOGOUT_SUCCESS',
  SET_USERNAME = 'SET_USERNAME',
  ALERT = 'ALERT',
  HIDE_ALERT = 'HIDE_ALERT',
  GET_USER = 'GET_USER',
  DEFAULT = 'DEFAULT',
  DICOM_PREVIEW_LOADED = 'DICOM_PREVIEW_LOADED',
  PDF_PREVIEW_LOADED = 'PDF_PREVIEW_LOADED',
  BATCH_PREVIEW_LOADED = 'BATCH_PREVIEW_LOADED',
  BATCH_PROCESS_LOADED = 'BATCH_PROCESS_LOADED',
  REVIEW_TAGS_LOADED = 'REVIEW_TAGS_LOADED',
}
export const CURRENT_DATE = new Date();
interface AuthActionType<T, P> {
  type: T;
  payload?: P;
}
export type AuthAction =
  | AuthActionType<typeof AuthActions.USER_LOADING, Auth>
  | AuthActionType<typeof AuthActions.USER_LOADED, undefined>
  | AuthActionType<typeof AuthActions.AUTH_ERROR, number>
  | AuthActionType<typeof AuthActions.LOGIN_SUCCESS, string>
  | AuthActionType<typeof AuthActions.LOGIN_FAIL, number>
  | AuthActionType<typeof AuthActions.LOGOUT_SUCCESS, undefined>
  | AuthActionType<typeof AuthActions.SET_USERNAME, string>
  | AuthActionType<typeof AuthActions.ALERT, string>
  | AuthActionType<typeof AuthActions.HIDE_ALERT, undefined>
  | AuthActionType<typeof AuthActions.GET_USER, any>
  | AuthActionType<typeof AuthActions.DEFAULT, undefined>
  | AuthActionType<typeof AuthActions.DICOM_PREVIEW_LOADED, any>
  | AuthActionType<typeof AuthActions.PDF_PREVIEW_LOADED, any>
  | AuthActionType<typeof AuthActions.BATCH_PREVIEW_LOADED, any>
  | AuthActionType<typeof AuthActions.BATCH_PROCESS_LOADED, any>
  | AuthActionType<typeof AuthActions.REVIEW_TAGS_LOADED, any>;
