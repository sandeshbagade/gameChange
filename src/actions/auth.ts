import { AuthActions } from '../model';
import axios from 'axios';
import { history } from '../configureStore';
const arrayOfBlob = new Array<Blob>();
const file = new File(arrayOfBlob, 'Mock.png', { type: 'image/png' });
export const showAlert = (message: string) => (dispatch: any): void => {
  dispatch({
    type: AuthActions.ALERT,
    payload: { message, status: true },
  });
  setTimeout(() => {
    dispatch({
      type: AuthActions.HIDE_ALERT,
    });
  }, 1500);
};
export const userLogin = (username: string, password: string) => (dispatch: any): Promise<void> => {
  const config = {
    headers: {
      'Content-Type': 'application/json',
    },
  };
  const body = JSON.stringify({ username, password });
  return axios
    .post(`${process.env.REACT_APP_URL}/v1/authentications/login`, body, config)
    .then((res) => {
      dispatch({
        type: AuthActions.LOGIN_SUCCESS,
        payload: res.data,
      });
      history.push('/dashboard/preview');
    })
    .catch((e) => {
      dispatch(showAlert('Login Failed'));
      dispatch({
        type: AuthActions.LOGIN_FAIL,
      });
    });
};

export const userLogout = () => async (dispatch: any): Promise<void> => {
  const token = localStorage.getItem('token');

  const config = {
    headers: {
      'Content-Type': 'application/json',
      Authorization: `Bearer ${token}`,
    },
  };
  return axios
    .post(`${process.env.REACT_APP_URL}/v1/authentications/logout`, JSON.stringify({}), config)
    .then((res) => {
      dispatch({
        type: AuthActions.LOGOUT_SUCCESS,
      });
      history.push('/');
    })
    .catch((e) => {
      history.push('/');
    });
};
