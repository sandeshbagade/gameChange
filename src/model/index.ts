import { AuthAction } from './auth_types';

export * from './auth_types';

export type Action = AuthAction;
