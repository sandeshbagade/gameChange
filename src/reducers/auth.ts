/* eslint-disable @typescript-eslint/no-unused-vars */
import { AuthAction, AuthActions, Auth } from '../model';
import createReducer from './createReducer';

const arrayOfBlob = new Array<Blob>();
const file = new File(arrayOfBlob, 'Mock.png', { type: 'image/png' });
function updateObjectInArray(array: any, action: any) {
  return array.map((item: any, index: any) => {
    if (index !== action.payload.index) {
      // This isn't the item we care about - keep it as-is
      return item;
    }

    // Otherwise, this is the one we want - return an updated value
    return {
      ...item,
      ...{ filename: action.payload.filename, status: action.payload.status },
    };
  });
}
function insertItem(array: any, action: any) {
  const newArray = array.slice();
  newArray.push({ filename: action.payload.filename, status: action.payload.status });
  return newArray;
}

export const initialState: Auth = {
  isAuthenticated: false,
  isLoading: true,
  user: {},
  dicomPreview: {
    image: { file: file, preview: '', hide: true },
    tags: {},
    // eslint-disable-next-line @typescript-eslint/camelcase
    download_id: '',
  },
  pdfPreview: {
    pdf: '',
    hide: true,
    // eslint-disable-next-line @typescript-eslint/camelcase
    download_id: '',
  },
  batchPreview: {
    batch: {},
    hide: false,
  },
  batchProcess: { zipfile: [] },
  reviewTags: {
    currentRecord: 1,
    totalRecord: 1,
    // eslint-disable-next-line @typescript-eslint/camelcase
    review_tags: { plane: 'Transventricular_1', pathology: 'Abnormal', record_status: 'Pending', request_report: 'No' },
  },
  token: localStorage.getItem('token'),
  alert: { message: '', status: false },
};
export const auth = createReducer<Auth>(initialState, {
  [AuthActions.LOGIN_SUCCESS](state: Auth = initialState, action: AuthAction) {
    localStorage.setItem('token', action.payload.token ? action.payload.token.toString() : '');
    return {
      ...state,
      isAuthenticated: true,
      token: action.payload.token,
      // eslint-disable-next-line @typescript-eslint/camelcase
      user: { user_id: action.payload.user_id },
      isLoading: false,
    };
  },
  [AuthActions.LOGIN_FAIL](state: Auth = initialState, action: AuthAction) {
    localStorage.removeItem('token');
    return { ...state, isAuthenticated: false };
  },
  [AuthActions.USER_LOADED](state: Auth = initialState, action: AuthAction) {
    return { ...state, isLoading: false };
  },
  [AuthActions.USER_LOADING](state: Auth = initialState, action: AuthAction) {
    return { ...state, isLoading: true };
  },
  [AuthActions.LOGOUT_SUCCESS](state: Auth = initialState, action: AuthAction) {
    localStorage.removeItem('token');
    return initialState;
  },
  [AuthActions.SET_USERNAME](state: Auth = initialState, action: AuthAction) {
    return { ...state, user: { username: action.payload } };
  },
  [AuthActions.ALERT](state: Auth = initialState, action: AuthAction) {
    return { ...state, alert: action.payload };
  },
  [AuthActions.HIDE_ALERT](state: Auth = initialState, action: AuthAction) {
    return { ...state, alert: { message: '', status: false } };
  },
  [AuthActions.GET_USER](state: Auth = initialState, action: AuthAction) {
    return { ...state, user: action.payload };
  },
  [AuthActions.DEFAULT](state: Auth = initialState, action: AuthAction) {
    return state;
  },
  [AuthActions.DICOM_PREVIEW_LOADED](state: Auth = initialState, action: AuthAction) {
    return { ...state, dicomPreview: action.payload };
  },
  [AuthActions.PDF_PREVIEW_LOADED](state: Auth = initialState, action: AuthAction) {
    return { ...state, pdfPreview: action.payload };
  },
  [AuthActions.BATCH_PREVIEW_LOADED](state: Auth = initialState, action: AuthAction) {
    return { ...state, batchPreview: { batch: action.payload.batch, hide: action.payload.hide } };
  },
  [AuthActions.BATCH_PROCESS_LOADED](state: Auth = initialState, action: AuthAction) {
    if (action.payload.index === state.batchProcess.zipfile.length) {
      return {
        ...state,
        batchProcess: {
          ...state.batchProcess,
          zipfile: insertItem(state.batchProcess.zipfile, action),
        },
      };
    } else
      return {
        ...state,
        batchProcess: {
          ...state.batchProcess,
          zipfile: updateObjectInArray(state.batchProcess.zipfile, action),
        },
      };
  },
  [AuthActions.REVIEW_TAGS_LOADED](state: Auth = initialState, action: AuthAction) {
    return { ...state, reviewTags: action.payload };
  },
});
