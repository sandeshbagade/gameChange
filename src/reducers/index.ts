import { History } from 'history';
import { combineReducers } from 'redux';
import { Auth } from '../model';
import * as authReducer from './auth';

export interface RootState {
  auth: Auth[];
}

export default (history: History) => {
  return combineReducers({
    auth: authReducer.auth,
  });
};
