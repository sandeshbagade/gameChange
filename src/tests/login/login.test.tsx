import React from 'react';
import Login from '../../login/login';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { MemoryRouter, Route } from 'react-router-dom';
import { initialState } from '../../reducers/auth';
import * as auth from '../../actions/auth';
import configureStore from '../mockConfigureStore';
import { Auth } from '../../model';
const MockInitialState: Auth = {
  ...initialState,
  isAuthenticated: false,
  isLoading: true,
  user: {},
  token: 'token',
};
const { store } = configureStore(MockInitialState);
describe('Login->', () => {
  it('Login snapshot test', () => {
    const link = renderer
      .create(
        <Provider store={store}>
          <Login />
        </Provider>,
      )
      .toJSON();
    expect(link).toMatchSnapshot();
  });
  it('should call userLogin action on click login button', () => {
    const actions = {
      userLogin: jest.spyOn(auth, 'userLogin'),
    };
    const wrapper = mount(
      <MemoryRouter initialEntries={['/']} initialIndex={0}>
        <Provider store={store}>
          <Route path="/" render={(): JSX.Element => <Login {...store} actions={actions} />} />
        </Provider>
      </MemoryRouter>,
    );
    const username = wrapper.find('[data-test-id="username"]').first().find('input');
    username.simulate('change', { target: { value: 'somevalue' } });
    const password = wrapper.find('[data-test-id="password"]').first().find('input');
    password.simulate('change', { target: { value: 'somevalue' } });
    const submitButton = wrapper.find('[data-test-id="loginButton"]').first();
    submitButton.simulate('click');
    expect(actions.userLogin).toHaveBeenCalled();
  });
});
