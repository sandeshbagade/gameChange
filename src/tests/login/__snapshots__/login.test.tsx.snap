// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login-> Login snapshot test 1`] = `
<div
  className="makeStyles-root-2"
>
  <div
    style={
      Object {
        "padding": 20,
        "textAlign": "center",
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-md-2 MuiGrid-grid-lg-2"
      >
        <img
          alt=""
          src="logo.png"
          style={
            Object {
              "height": "80px",
            }
          }
        />
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-md-10 MuiGrid-grid-lg-10"
        style={
          Object {
            "color": "#333333",
            "fontFamily": "Helvetica Neue",
            "margin": "55px 0 0 -4px",
          }
        }
      >
        <span
          style={
            Object {
              "fontSize": 24,
            }
          }
        >
          ORIGIN HEALTH VAULT
        </span>
        <sup>
          TM
        </sup>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-cardRoot-1 MuiPaper-elevation3 MuiPaper-rounded"
  >
    <div
      className="MuiCardContent-root"
    >
      <p
        className="MuiTypography-root MuiTypography-h6"
        style={
          Object {
            "marginBottom": "-10px",
            "padding": "20px 0",
          }
        }
      >
        Username
      </p>
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        data-test-id="username"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={true}
            className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            disabled={false}
            id="username"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-6"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
      <p
        className="MuiTypography-root MuiTypography-h6"
        style={
          Object {
            "marginBottom": "-15px",
            "padding": "20px 0",
          }
        }
      >
        Password
      </p>
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        data-test-id="password"
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            disabled={false}
            id="password"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="password"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-6"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
      <br />
      <br />
      <div
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-customButton-3 MuiButton-disableElevation"
          data-test-id="loginButton"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "marginBottom": "30px",
            }
          }
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <p
              className="MuiTypography-root MuiTypography-h6"
            >
              <span
                style={
                  Object {
                    "padding": "2px 5px 0 5px",
                    "textTransform": "capitalize",
                  }
                }
              >
                Login
              </span>
            </p>
          </span>
        </button>
        <br />
        <br />
        <span
          className="MuiTypography-root MuiTypography-subtitle2"
          onClick={[Function]}
        >
          <span
            className="makeStyles-passwordResetLink-4"
          >
            Request password reset
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
